src/test/llmnode_test.cpp

```diff
@@ -101,32 +135,524 @@ class LLMOptionsHttpTest : public ::testing::TestWithParam<std::string> {
 // with user defined one to do that.
 // TODO: Consider stress testing - exisitng model server under heavy load to check notifications work us expected.
 
-// Test disabled by default - needs LLM models to work in /workspace directory:
-// openvino_detokenizer.bin  openvino_detokenizer.xml  openvino_model.bin  openvino_model.xml  openvino_tokenizer.bin  openvino_tokenizer.xml
-// Model that was used:
-// https://huggingface.co/meta-llama/Llama-2-7b-chat-hf
-// converted with optimum cli
-
-// TODO: Switch to HTTP caclulator
-TEST_F(LLMFlowHttpTest, DISABLED_Infer) {
-    const ovms::Module* grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
-    KFSInferenceServiceImpl& impl = dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
-    ::KFSRequest request;
-    ::KFSResponse response;
-    const std::string modelName = "llmDummyKFS";
-    request.Clear();
-    response.Clear();
-    std::vector<std::string> requestData1{"What is OpenVINO?"};
-    std::string expectedResponse = "\n\nOpenVINO is an open-source software library for deep learning inference that is designed to optimize and run deep learning models on a variety";
-
-    prepareInferStringRequest(request, "in", requestData1, false);
-    request.mutable_model_name()->assign(modelName);
-    ASSERT_EQ(impl.ModelInfer(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
-    const std::string& content = response.raw_output_contents(0);
-    ASSERT_EQ(expectedResponse, content);
-}
-
-TEST_F(LLMFlowHttpTest, LLMNodeNameMissing) {
+TEST_F(LLMFlowHttpTest, inferChatCompletionsUnary) {
+    std::string requestBody = R"(
+        {
+            "model": "llmDummyKFS",
+            "stream": false,
+            "seed" : 1,
+            "max_tokens": 5,
+            "messages": [
+            {
+                "role": "user",
+                "content": "What is OpenVINO?"
+            }
+            ]
+        }
+    )";
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::OK);
+    // Assertion split in two parts to avoid timestamp missmatch
+    const size_t timestampLength = 10;
+    std::string expectedResponsePart1 = R"({"choices":[{"finish_reason":"stop","index":0,"logprobs":null,"message":{"content":"\nOpenVINO is","role":"assistant"}}],"created":)";
+    std::string expectedResponsePart2 = R"(,"model":"llmDummyKFS","object":"chat.completion"})";
+    ASSERT_EQ(response.compare(0, expectedResponsePart1.length(), expectedResponsePart1), 0);
+    ASSERT_EQ(response.compare(expectedResponsePart1.length() + timestampLength, expectedResponsePart2.length(), expectedResponsePart2), 0);
```

Comment:
I think it's risky to compare output of a real model with fixed text in oncommit tests. This can fail of different hardware due to some internal math computation differences. I suppose we do that in more places.